src/services/blocks/XCMDecoder.ts

```diff
@@ -0,0 +1,158 @@
+// Copyright 2017-2024 Parity Technologies (UK) Ltd.
+// This file is part of Substrate API Sidecar.
+//
+// Substrate API Sidecar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import '@polkadot/api-augment';
+
+import { ApiPromise } from '@polkadot/api';
+import { Bytes } from '@polkadot/types';
+import { ISanitizedBackedCandidate } from 'src/types/responses/SanitizedBackedCandidate';
+import { ISanitizedParachainInherentData } from 'src/types/responses/SanitizedParachainInherentData';
+import { ISanitizedParentInherentData } from 'src/types/responses/SanitizedParentInherentData';
+
+import {
+	IDownwardMessage,
+	IExtrinsic,
+	IFrameMethod,
+	IHorizontalMessage,
+	IMessages,
+	IUpwardMessage,
+} from '../../types/responses';
+
+enum ChainType {
+	Relay = 'Relay',
+	Parachain = 'Parachain',
+}
+
+export class XcmDecoder {
```

Comment:
As noted in the below corrections, in this case we should have the methods not be static, and be private internal to the instance that is created so we can reference them using `this`.

The initial doc we created has some contradictions,  because we wrote static, but then referenced the methods using `this`. In this case there should be no static in the class.

Reply:
The bottoms corrects are just top level, their might need to be changes internal to the functions themselves as well.